server-slicer/utils.py

```diff
@@ -0,0 +1,1040 @@
+                    # move the file to the cache directory
+                    file_name = os.path.basename(item)
+                    self.cache_images[item] = os.path.join(self.cache_dir, file_name)
+                    if not os.path.isfile(self.cache_images[item]):
+                        copyfile(item, self.cache_images[item])
+
+                if self.cache_images[item].endswith(".nii.gz"):
+                    data = nib.load(self.cache_images[item]).get_fdata()
+                    for slice_index in tqdm(range(data.shape[2])):
+                        image_filename = get_slice_filenames(self.cache_images[item], slice_index)
+                        if not os.path.exists(os.path.join(self.cache_dir, image_filename)):
+                            compose = get_monai_transforms(
+                                ["image"],
+                                self.cache_dir,
+                                modality=get_modality(item),
+                                slice_index=slice_index,
+                                image_filename=image_filename,
+                            )
+                            compose({"image": self.cache_images[item]})
+
+    def cleanup(self):
+        """Clean up the cache directory."""
+        logger.debug(f"Cleaning up the cache")
+        rmtree(self.cache_dir)
+
+    def dir(self):
+        """Return the cache directory."""
+        return str(self.cache_dir)
+
+    def get(self, key: str | list, default=None, list_return=False):
+        """Get the image or data URL from the cache."""
+        if isinstance(key, list):
+            items = [self.cache_images.get(k) for k in key]
+            return items if list_return else items[0]
+        return self.cache_images.get(key, default)
+
+
+class SessionVariables:
+    """Class to store the session variables"""
+
+    def __init__(self):
+        """Initialize the session variables"""
+        global SYS_PROMPT
+        self.sys_prompt = SYS_PROMPT
+        self.sys_msg = MODEL_CARDS
+        self.use_model_cards = True
+        self.slice_index = None  # Slice index for 3D images
+        self.image_url = None  # Image URL to the image on the web
+        self.backup = {}  # Cached varaiables from previous messages for the current conversation
+        self.axis = 2
+        self.top_p = 0.9
+        self.temperature = 0.0
+        self.max_tokens = 1024
+        self.temp_working_dir = None
+        # self.idx_range = (None, None)
+        # self.interactive = False
+        self.sys_msgs_to_hide = []
+        self.modality_prompt = "Auto"
+
+    def restore_from_backup(self, attr):
+        """Retrieve the attribute from the backup"""
+        attr_val = self.backup.get(attr, None)
+        if attr_val is not None:
+            self.__setattr__(attr, attr_val)
+
+
+class ExpertVista3D():
+    """Expert model for VISTA-3D."""
+
+    NIM_VISTA3D = "https://api.nvcf.nvidia.com/v2/nvcf/pexec/functions/caf65f74-1c04-4b3c-b566-5908ee950f24"
```

Comment:
Hi @diazandr3s 
This is an internal NVCF. Having it exposed in the code is not intended.